lib/api.js

```diff
@@ -1035,50 +1042,47 @@ API.prototype.createWalletFromOldCopay = function(username, password, blob, cb)
   var walletName = w.opts.name;
   var network = w.opts.networkName;
   this.credentials = Credentials.fromOldCopayWallet(w);
-
   var walletPrivKey = this._walletPrivKeyFromOldCopayWallet(w);
+  var copayerName = this.credentials.copayerName;
 
-
-  // Grab My Copayer Name
-  var hd = new Bitcore.HDPublicKey(self.credentials.xPubKey).derive('m/2147483646/0/0');
-  var pubKey = hd.publicKey.toString('hex');
-  var copayerName = w.publicKeyRing.nicknameFor[pubKey] || username;
-
-
-  this.createWallet(walletName, copayerName, m, n, {
-    network: network,
-    id: walletId,
-    walletPrivKey: walletPrivKey,
-  }, function(err, secret) {
-
-    if (err && err.code == 'WEXISTS') {
-
+  // First: Try to get the wallet with the imported credentials...
+  this.getStatus(function(err) {
+    // No error? -> Wallet is ready.
+    if (!err) {
+      log.debug('Wallet is already imported');
       self.credentials.addWalletInfo(walletId, walletName, m, n,
         walletPrivKey, copayerName);
+      return cb();
+    };
 
-      return self._replaceTemporaryRequestKey(function(err) {
-        if (err) return cb(err);
-        self.openWallet(function(err) {
-          return cb(err, true);
+    self.createWallet(walletName, copayerName, m, n, {
+      network: network,
+      id: walletId,
+      walletPrivKey: walletPrivKey,
+    }, function(err, secret) {
+      if (err && err.code == 'WEXISTS') {
+
+        self.credentials.addWalletInfo(walletId, walletName, m, n,
```

Comment:
This is a bit paranoid but shouldn't we check our copayerId is included in this wallet before we save the credentials?